design-proposals/flavors-and-preferences.md

```diff
@@ -0,0 +1,1023 @@
+        * `VirtualMachinePreferenceSpec.Firmware.PreferredUseBios`
+        * useSerial
+          * `VirtualMachinePreferenceSpec.Firmware.PreferredUseBiosSerial`
+      * [efi](http://kubevirt.io/api-reference/main/definitions.html#_v1_efi)
+        * `VirtualMachinePreferenceSpec.Firmware.PreferredUseEfi`
+        * secureBoot
+          * `VirtualMachinePreferenceSpec.Firmware.PreferredUseEfiSecureBoot`
+    * [kernelBoot](http://kubevirt.io/api-reference/main/definitions.html#_v1_kernelboot)
+      * [container](http://kubevirt.io/api-reference/main/definitions.html#_v1_kernelbootcontainer)
+        * image
+          * None
+        * imagePullPolicy
+          * None
+        * imagePullSecret
+          * None
+        * initrdPath
+          * None
+        * kernelPath
+          * None
+      * kernelArgs
+        * None
+    * serial
+      * None
+    * uuid
+      * None
+  * ioThreadsPolicy
+    * `VirtualMachineFlavorSpec.ioThreadsPolicy`
+  * [launchSecurity](http://kubevirt.io/api-reference/main/definitions.html#_v1_launchsecurity)
+    * [sev](http://kubevirt.io/api-reference/main/definitions.html#_v1_sev)
+      * `VirtualMachineFlavorSpec.LaunchSecurity.SEV`
+  * [Machine](http://kubevirt.io/api-reference/main/definitions.html#_v1_machine)
+    * type
+      * `VirtualMachinePreferenceSpec.Machine.PreferredMachineType`
+  * [Memory](http://kubevirt.io/api-reference/main/definitions.html#_v1_memory)
+    * [guest](http://kubevirt.io/api-reference/main/definitions.html#_k8s_io_apimachinery_pkg_api_resource_quantity) 
+      * `VirtualMachineFlavorSpec.Memory.Guest`
+    * [hugepages](http://kubevirt.io/api-reference/main/definitions.html#_v1_hugepages)
+      * `VirtualMachineFlavorSpec.Memory.HugePages`
+
+## Types
+
+### `VirtualMachineFlavorSpec`
+
+```go
+// VirtualMachineFlavorSpec
+//
+// +k8s:openapi-gen=true1
+type VirtualMachineFlavorSpec struct {
+	CPU *CPUFlavor `json:"cpu"`
+
+	Memory *MemoryFlavor `json:"memory"`
+
+	// Optionally defines any GPU devices associated with the flavor.
+	//
+	// +optional
+	// +listType=atomic
+	GPUs []v1.GPU `json:"gpus,omitempty"`
+
+	// Optionally defines any HostDevices associated with the flavor.
+	//
+	// +optional
+	// +listType=atomic
+	HostDevices []v1.HostDevice `json:"hostDevices,omitempty"`
+
+	// Optionally defines the IOThreadsPolicy to be used by the flavor.
+	//
+	// +optional
+	IOThreadsPolicy *v1.IOThreadsPolicy `json:"ioThreadsPolicy,omitempty"`
+
+	// Optionally defines the LaunchSecurity to be used by the flavor.
+	//
+	// +optional
+	LaunchSecurity *v1.LaunchSecurity `json:"launchSecurity,omitempty"`
+}
+```
+
+### `CPUFlavor`
+
+```go
+// CPUFlavor
+//
+// +k8s:openapi-gen=true
+type CPUFlavor struct {
+
+	// Number of vCPUs to expose to the guest.
+	// The resulting CPU topology being derived from the optional PreferredCPUTopology attribute of CPUPreferences.
+	Guest uint32 `json:"guest"`
```

Comment:
Same here: We should include a unit in the field name

Reply:
ACK.